### PR TITLE
chore: use testify instead of testing.Fatal or testing.Error in applicationset

### PR DESCRIPTION
**Pull request by mmorel-35**
### Description

* uses testify instead of testing.Fatal or testing.Error in applicationset

### CODE REVIEW

File: applicationset/controllers/applicationset_controller_test.go [hunk 1]
```diff
@@ -1178,18 +1178,18 @@ func TestRemoveFinalizerOnInvalidDestination_FinalizerTypes(t *testing.T) {
 			// argoDB := db.NewDB("namespace", settingsMgr, r.KubeClientset)
 			// clusterList, err := argoDB.ListClusters(context.Background())
 			clusterList, err := utils.ListClusters(context.Background(), kubeclientset, "namespace")
-			require.NoError(t, err, "Unexpected error")
+			require.NoErrorf(t, err, "Unexpected error")
```
Comment (andrii-korotkov-verkada) [inline applicationset/controllers/applicationset_controller_test.go:1181]: These changes seem unnecessary.

Comment (andrii-korotkov-verkada) [reply applicationset/controllers/applicationset_controller_test.go:1181]: Please, revert the changes to places which don't have any args.

File: applicationset/services/pull_request/github_test.go [hunk 2]
```diff
@@ -52,9 +52,8 @@ func TestContainLabels(t *testing.T) {
 
 	for _, c := range cases {
 		t.Run(c.Name, func(t *testing.T) {
-			if got := containLabels(c.Labels, c.PullLabels); got != c.Expect {
-				t.Errorf("expect: %v, got: %v", c.Expect, got)
-			}
+			got := containLabels(c.Labels, c.PullLabels)
+			assert.Equal(t, got, c.Expect, "expect: %v, got: %v", c.Expect, got)
```
Comment (andrii-korotkov-verkada) [inline applicationset/services/pull_request/github_test.go:56]: Should this use require as well?

File: applicationset/services/pull_request/gitlab_test.go [hunk 3]
```diff
@@ -17,13 +17,10 @@ import (
...
-	if err != nil {
-		t.Fatalf("error opening fixture file: %v", err)
-	}
+	require.NoError(t, err, "error opening fixture file: %v", err)
 
-	if _, err = io.Copy(w, f); err != nil {
-		t.Fatalf("error writing response: %v", err)
-	}
+	_, err = io.Copy(w, f)
+	require.NoError(t, err, "error writing response: %v", err)
```
Comment (andrii-korotkov-verkada) [inline applicationset/services/pull_request/gitlab_test.go:23]: This probably should be `NoErrorf`

File: applicationset/services/pull_request/gitlab_test.go [hunk 4]
```diff
@@ -17,13 +17,10 @@ import (
 func writeMRListResponse(t *testing.T, w io.Writer) {
 	t.Helper()
 	f, err := os.Open("fixtures/gitlab_mr_list_response.json")
-	if err != nil {
-		t.Fatalf("error opening fixture file: %v", err)
-	}
+	require.NoError(t, err, "error opening fixture file: %v", err)
```
Comment (andrii-korotkov-verkada) [inline applicationset/services/pull_request/gitlab_test.go:20]: This probably should be `NoErrorf`

File: applicationset/webhook/webhook_test.go [hunk 5]
```diff
@@ -242,9 +242,8 @@ func TestWebhookHandler(t *testing.T) {
 			for i := range list.Items {
 				gotAppSet := &list.Items[i]
 				if _, isEffected := effectedAppSetsAsExpected[gotAppSet.Name]; isEffected {
-					if expected, got := test.expectedRefresh, gotAppSet.RefreshRequired(); expected != got {
-						t.Errorf("unexpected RefreshRequired() for appset '%s' expect: %v got: %v", gotAppSet.Name, expected, got)
-					}
+					expected, got := test.expectedRefresh, gotAppSet.RefreshRequired()
+					assert.Equal(t, expected, got, "unexpected RefreshRequired() for appset '%s' expect: %v got: %v", gotAppSet.Name, expected, got)
```
Comment (andrii-korotkov-verkada) [inline applicationset/webhook/webhook_test.go:246]: Should we use require here as well?

File: applicationset/services/plugin/plugin_service_test.go [hunk 6]
```diff
@@ -31,19 +32,13 @@ func TestPlugin(t *testing.T) {
...
-	if err != nil {
-		t.Errorf("unexpected error: %v", err)
-	}
+	require.NoError(t, err, "unexpected error")
 
 	data, err := client.List(context.Background(), nil)
-	if err != nil {
-		t.Errorf("unexpected error: %v", err)
-	}
+	require.NoError(t, err, "unexpected error")
```
Comment (andrii-korotkov-verkada) [inline applicationset/services/plugin/plugin_service_test.go:38]: We probably don't need that extra "unexpected error" param, as default message might be sufficient.

File: applicationset/services/internal/http/client_test.go [hunk 7]
```diff
@@ -149,12 +144,7 @@ func TestCheckResponse(t *testing.T) {
...
-	if err == nil {
-		t.Error("Expected an error, got nil")
-	}
+	require.Error(t, err)
 
-	expected := "API error with status code 400: invalid_request"
-	if err.Error() != expected {
-		t.Errorf("Expected error '%s', got '%s'", expected, err.Error())
-	}
+	assert.EqualError(t, err, "API error with status code 400: invalid_request")
```
Comment (andrii-korotkov-verkada) [inline applicationset/services/internal/http/client_test.go:149]: Use require.

File: applicationset/services/pull_request/github_test.go [hunk 8]
```diff
@@ -52,9 +52,8 @@ func TestContainLabels(t *testing.T) {
 
 	for _, c := range cases {
 		t.Run(c.Name, func(t *testing.T) {
-			if got := containLabels(c.Labels, c.PullLabels); got != c.Expect {
-				t.Errorf("expect: %v, got: %v", c.Expect, got)
-			}
+			got := containLabels(c.Labels, c.PullLabels)
+			assert.Equal(t, got, c.Expect)
```
Comment (andrii-korotkov-verkada) [inline applicationset/services/pull_request/github_test.go:56]: Use require

**Review by andrii-korotkov-verkada — APPROVED** [review]

**Review by andrii-korotkov-verkada — APPROVED** [review]

**Review by crenshaw-dev — APPROVED** [review]

**Pull request merged**